frame/session/src/lib.rs

```diff
@@ -663,42 +662,31 @@ impl<T: Config> Module<T> {
 		T::SessionHandler::on_new_session::<T::Keys>(changed, &session_keys, &queued_amalgamated);
 	}
 
-	/// Disable the validator of index `i`.
-	///
-	/// Returns `true` if this causes a `DisabledValidatorsThreshold` of validators
-	/// to be already disabled.
+	/// Disable the validator of index `i`, returns `false` if the validator was already disabled.
 	pub fn disable_index(i: usize) -> bool {
-		let (fire_event, threshold_reached) = DisabledValidators::mutate(|disabled| {
-			let i = i as u32;
-			if let Err(index) = disabled.binary_search(&i) {
-				let count = <Validators<T>>::decode_len().unwrap_or(0) as u32;
-				let threshold = T::DisabledValidatorsThreshold::get() * count;
-				disabled.insert(index, i);
-				(true, disabled.len() as u32 > threshold)
-			} else {
-				(false, false)
+		DisabledValidators::mutate(|disabled| {
```

Comment:
To clarify my understanding, is the purpose of the `DisabledValidators` storage item just to prevent unnescary calls to `T::SessionHandler::on_disabled(i)` ?

Reply:
I think I see: it will be used in babe/aura like https://github.com/paritytech/substrate/pull/9414/files#diff-f5430be578eba5004ae7f2f75f95f5ce7a2819be7babb0f7f0477cd027b35facR896-R900

Reply:
I'm not against using a set here but I just kept the original data structure and we're using binary search to search this vec.